plugins/sonarqube/src/api/SonarQubeClient.ts

```diff
@@ -34,20 +35,37 @@ export class SonarQubeClient implements SonarQubeApi {
     this.baseUrl = baseUrl.endsWith('/') ? baseUrl : `${baseUrl}/`;
   }
 
-  private async callApi<T>(path: string): Promise<T | undefined> {
+  private async callApi<T>(
+    path: string,
+    query?: { [key in string]: any },
+  ): Promise<T | undefined> {
+    const queryString = query ? `?${qs.stringify(query)}` : '';
```

Comment:
Would `URLSearchParams` be fine here instead of `qs`? `new URLSearchParams(undefined).toString()` gives you `''` too

Reply:
Yes that also works. I mainly wanted to get rid of the trailing `?` but that is not really necessary since it is still a valid URL. I'll change it.

Reply:
Just realized that the `undefined` case isn't used at all.